ffi/android/Dockerfile

```diff
@@ -0,0 +1,60 @@
+FROM rustlang/rust:nightly-slim as builder
+
+# mkdir required to install openjdk on slim images
+RUN mkdir -p /usr/share/man/man1 && \
+    apt-get update && \
+    apt-get install -y \
+        cmake g++ git libpcre3-dev make openjdk-11-jdk unzip wget && \
+    apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+WORKDIR /tmp
+
+RUN wget https://freefr.dl.sourceforge.net/project/swig/swig/swig-4.0.1/swig-4.0.1.tar.gz && \
+    tar xf swig-*.tar.gz && \
+    cd swig-* && \
+    ./configure && make -j12 && \
+    make install && \
+    rm -rf /tmp/*
+
+ENV ANDROID_SDK_ROOT=/root/sdk
+ENV NDK_VERSION=20.1.5948944
+ENV NDK_HOME=$ANDROID_SDK_ROOT/ndk/$NDK_VERSION
+
+RUN wget https://dl.google.com/android/repository/commandlinetools-linux-6609375_latest.zip && \
+    unzip commandlinetools-linux-*_latest.zip && \
+    mkdir $ANDROID_SDK_ROOT && \
+    mv tools $ANDROID_SDK_ROOT/ && \
+    rm -rf /tmp/*
+
+RUN yes | $ANDROID_SDK_ROOT/tools/bin/sdkmanager --sdk_root=$ANDROID_SDK_ROOT \
+        "platform-tools" \
+        "build-tools;29.0.3" \
+        "platforms;android-29" \
+        "ndk;$NDK_VERSION" \
+        "cmake;3.10.2.4988404"
+
+RUN echo '[target.aarch64-linux-android]\n\
+ar = "/root/sdk/ndk/20.1.5948944/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android-ar"\n\
+linker = "/root/sdk/ndk/20.1.5948944/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android26-clang"\n\n\
+[target.x86_64-linux-android]\n\
+ar = "/root/sdk/ndk/20.1.5948944/toolchains/llvm/prebuilt/linux-x86_64/bin/x86_64-linux-android-ar"\n\
+linker = "/root/sdk/ndk/20.1.5948944/toolchains/llvm/prebuilt/linux-x86_64/bin/x86_64-linux-android26-clang"\n\n\
+[target.armv7-linux-androideabi]\n\
+ar = "/root/sdk/ndk/20.1.5948944/toolchains/llvm/prebuilt/linux-x86_64/bin/arm-linux-androideabi-ar"\n\
+linker = "/root/sdk/ndk/20.1.5948944/toolchains/llvm/prebuilt/linux-x86_64/bin/armv7a-linux-androideabi26-clang"\n\n\
+[target.i686-linux-android]\n\
+ar = "/root/sdk/ndk/20.1.5948944/toolchains/llvm/prebuilt/linux-x86_64/bin/i686-linux-android-ar"\n\
+linker = "/root/sdk/ndk/20.1.5948944/toolchains/llvm/prebuilt/linux-x86_64/bin/i686-linux-android26-clang"' > $CARGO_HOME/config.toml
```

Comment:
May be better to create a file with this settings and later just copy it?

Reply:
from [here](https://doc.rust-lang.org/cargo/reference/config.html#hierarchical-structure) I understand that this config could be kept in the project. so I was about to propose to add it to the project in `ffi/android/.cargo/config.toml` and document on the local instructions to just update that file with the correct path to their `NDK_HOME`. but that solution wasn't really nice, since users, after changing the file with their paths, would always find the git working directory dirty

so reading documentation I've discovered 2 things:
- `ar` is deprecated, as documented [here](https://doc.rust-lang.org/cargo/reference/config.html#target) (I've tested it and I can confirm that it successfully builds without it)
- we can use configuration environment variables instead of cargo's config (documented [here](https://doc.rust-lang.org/cargo/reference/environment-variables.html#configuration-environment-variables))

this last solution is my preferred, I've added a commit with it so you can review it and tell me what do you think about it